### PR TITLE
10106 Story

### PR DESCRIPTION
**Pull request by codyseibert**
Refactor the mobile and public docket record header view to allow sorting of the entries

### CODE REVIEW

File: shared/src/business/useCases/getCaseInteractor.ts [hunk 1]
```diff
@@ -86,15 +87,20 @@ const getCaseForExternalUser = ({
...
  * @returns {Object} decorated caseRecord
  */
-export const decorateForCaseStatus = (caseRecord: Case) => {
+export const decorateForCaseStatus = <T extends RawCase | SealedCase>(
+  caseRecord: T,
+) => {
   // allow document service
-  caseRecord.canAllowDocumentService =
+  (caseRecord as any).canAllowDocumentService =
     canAllowDocumentServiceForCase(caseRecord);
```
Comment (TomElliottFlexion) [inline shared/src/business/useCases/getCaseInteractor.ts:92]: Would you consider making a new type, adding `canAllowDocumentService` to SealedCase, or (maybe?) removing SealedCase as a type from this function?

File: shared/src/business/useCases/getConsolidatedCasesByCaseInteractor.ts [hunk 2]
```diff
@@ -67,7 +67,7 @@ export const getConsolidatedCasesByCaseInteractor = async (
       const formattedPublicCase = formatPublicCase({
         applicationContext,
         rawCaseRecord: consolidatedCase,
-      });
+      }) as any;
       validatedConsolidatedCases.push(formattedPublicCase);
```
Comment (TomElliottFlexion) [inline shared/src/business/useCases/getConsolidatedCasesByCaseInteractor.ts]: Can this be of type `RawPublicCase`?

File: web-client/src/presenter/computeds/Public/publicCaseDetailHelper.ts [hunk 3]
```diff
@@ -201,19 +201,16 @@ const filterDocketEntries = (
   }
 };
 
-export interface IPublicCaseDetailHelper {
-  formattedDocketEntriesOnDocketRecord: any[];
```
Comment (TomElliottFlexion) [inline web-client/src/presenter/computeds/Public/publicCaseDetailHelper.ts:205]: Does this interface need to be removed?

File: web-client/src/views/DocketRecord/DocketRecordHeader.tsx [hunk 4]
```diff
@@ -1,13 +1,147 @@
...
+import { DocketRecordSort } from './DocketRecordSort';
 import { Mobile, NonMobile } from '../../ustc-ui/Responsive/Responsive';
 import { OpenPrintableDocketRecordModal } from './OpenPrintableDocketRecordModal';
 import { connect } from '@cerebral/react';
 import { sequences } from '@web-client/presenter/app.cerebral';
 import { state } from '@web-client/presenter/app.cerebral';
 import React from 'react';
 
+export const DocketRecordMobileHeader = ({
+  docketNumber,
```
Comment (TomElliottFlexion) [inline web-client/src/views/DocketRecord/DocketRecordHeader.tsx:12]: Nice!

**Review by TomElliottFlexion — APPROVED** [review]
Looks good!

File: web-client/src/presenter/computeds/Public/publicCaseDetailHelper.ts (hunk 3 above)
Comment (codyseibert) [reply web-client/src/presenter/computeds/Public/publicCaseDetailHelper.ts:205]: yes, it's not needed.  We found a better way to get types without needing to explicitly define this interface

**Review by mmarcotte — APPROVED** [review]

**Pull request merged**